istioctl/cmd/proxyconfig.go

```diff
@@ -1277,24 +1278,37 @@ func secretConfigCmd() *cobra.Command {
 			return nil
 		},
 		RunE: func(c *cobra.Command, args []string) error {
-			var configWriter *configdump.ConfigWriter
+			var newWriter writer.ConfigDumpWriter
 			var err error
 			if len(args) == 1 {
 				if podName, podNamespace, err = getPodName(args[0]); err != nil {
 					return err
 				}
-				configWriter, err = setupPodConfigdumpWriter(podName, podNamespace, false, c.OutOrStdout())
+				if isZtunnelPod(podName) {
+					newWriter, err = setupZtunnelConfigDumpWriter(podName, podNamespace, c.OutOrStdout())
+				} else {
+					newWriter, err = setupPodConfigdumpWriter(podName, podNamespace, false, c.OutOrStdout())
+				}
 			} else {
-				configWriter, err = setupFileConfigdumpWriter(configDumpFile, c.OutOrStdout())
+				newWriter, err = setupFileConfigdumpWriter(configDumpFile, c.OutOrStdout())
+				if err != nil {
+					envoyError := err
+					newWriter, err = setupFileZtunnelConfigdumpWriter(configDumpFile, c.OutOrStdout())
+					if err != nil {
+						// failed to parse envoy and ztunnel formats
+						log.Warnf("couldn't parse envoy secrets dump: %v", envoyError)
```

Comment:
Do we need to print two logs here? Maybe just one log like `couldn't parse proxy secrets dump`?

Reply:
At this point we don't know what was the intended dump input type: envoy or ztunnel.  So I wanted to make sure both errors were returned to user .  Otherwise only the envoy parse error is returned below. I am open to suggestions - I could wrap it in `multierror` if that is better?

Reply:
I think using `multierror` is the same. I'm open to the formats as well, let's just leave it like this.